### DISCUSSION TITLE
Can't load Spacy project model rel_component

### DISCUSSION THREAD

**Discussion by Onyoursix** [explosion/spaCy#7106] [Help: Coding & Implementations] [answered]
I'm super interested in some of the Spacy projects and using them to learn more. I downloaded the `tutorials/rel_component` project and trained it on the data in the assets, however when I try to load the model via`nlp = spacy.load('./training/model-best')` I get the following error.

ValueError: [E002] Can't find factory for 'relation_extractor' for language English (en). This usually happens when spaCy calls `nlp.create_pipe` with a custom component name that's not registered on the current language class. If you're using a Transformer, make sure to install 'spacy-transformers'. If you're using a custom component, make sure you've added the decorator `@Language.component` (for function components) or `@Language.factory` (for class components).

Available factories: attribute_ruler, tok2vec, merge_noun_chunks, merge_entities, merge_subtokens, token_splitter, parser, beam_parser, entity_linker, ner, beam_ner, entity_ruler, lemmatizer, tagger, morphologizer, senter, sentencizer, textcat, textcat_multilabel, en.lemmatizer

I'm not entirely sure how to resolve this error. Any ideas?

**Comment by svlandeg** [comment, marked as answer]
The `tutorials/rel_component` project defines a few new [custom functions](https://spacy.io/usage/training#custom-functions), that need to be available when you run the trained model. What happens behind the scenes is that spaCy loads the model from file with `spacy.load` and at that point reads in the [configuration file](https://spacy.io/usage/training#config) to build up the objects in memory. But the configuration file refers to those custom functions from the project, so you need to make sure they've been imported or are otherwise available to the [registry](https://spacy.io/api/top-level#registry) in your runtime environment.

If you import https://github.com/explosion/projects/blob/v3/tutorials/rel_component/scripts/custom_functions.py you'll probably get this resolved.

**Reply by Onyoursix** [reply]
That did it! Thank you! I'm super excited to dive more into this and learn more about Spacy. Also a big thank you for putting together that YouTube video and the rel_component tutorial!

**Reply by svlandeg** [reply]
Happy to hear it works, and that you've found the video/tutorial useful! 😊
